torch/csrc/jit/subgraph_matcher.cpp

```diff
@@ -140,6 +141,33 @@ bool SubgraphMatcher::matchNodes(const Node* n1, Node* n2) {
       return false;
     }
   }
+  std::vector<Symbol> attr_names = n1->attributeNames();
+  for (size_t i = 0; i < attr_names.size(); i++) {
+    auto attrname = attr_names[i];
+    if (n1->kindOf(attrname) != n2->kindOf(attrname)) {
+      return false;
+    }
+    switch (n1->kindOf(attrname)) {
+      case AttributeKind::s:
+        if (n1->s(attrname) != n2->s(attrname)) {
+          return false;
+        }
+        break;
+      case AttributeKind::f:
+        if (n1->f(attrname) != n2->f(attrname)) {
+          return false;
+        }
+        break;
+      case AttributeKind::i:
+        if (n1->i(attrname) != n2->i(attrname)) {
+          return false;
+        }
+        break;
+      default:
+        // Other attributes types not supported yet
```

Comment:
Should this error? Or is this not reachable?

Reply:
If we see an attribute of unknown type, we just conservatively don't match. It is reachable, but I'm not sure if we want to hard fail here.